suite/tests/client-interface/drbbdup-emul-test.dll.c

```diff
@@ -125,6 +126,30 @@ instrument_instr(void *drcontext, void *tag, instrlist_t *bb, instr_t *instr,
         }
     }
 
+#ifdef X86
+    /* Cause drutil's expanded "rep movs" to crash if drbbdup sets
+     * DRREG_IGNORE_CONTROL_FLOW (i#5398).  We do not spill rcx until after the
+     * rcx==0 path jumps over the OP_movs, such that drreg's restore before the
+     * OP_loop writes the wrong value.
+     */
+    if (instr_get_opcode(instr) == OP_movs) {
+        reg_id_t reg_clobber;
```

Comment:
There does not seem to be a global check for sanity, ensuring that we entered this branch in our test-case.

Reply:
Will add -- but it won't ensure we hit the crash which depends on a zero-iter rep movs loop.  Maybe it's worth adding a clean call or sthg to check that.  Note that the 32-bit simple_app happens to have a 0-iter loop so emul-test did in fact hit the crash there but it's just luck (and the 64-bit doesn't have one).

Reply:
We can always add an .asm test app like other tests no?

Reply:
Added both.

Reply:
> We can always add an .asm test app like other tests no?

I have inline asm in the new test app.  I wanted something larger than pure asm to cover a variety of block sizes for the emulation marker leak test.